backend/wmg/pipeline/integrated_corpus/extract.py

```diff
@@ -37,26 +24,41 @@ def get_X_raw(anndata_object: anndata.AnnData) -> Union[np.ndarray, sparse.spmat
     return raw_expression_matrix if raw_expression_matrix is not None else anndata_object.X
 
 
-def get_dataset_s3_uris() -> Dict[str, str]:
+def get_dataset_s3_uris(datasets=None) -> Dict[str, str]:
     """
     Retrieve list of s3 uris for datasets included in the wmg cube
+
+    :param datasets: list of datasets to check, if None, will retrieve from API
+        This parameter is used for tests.
     """
+    # hardcode to dev backend if deployment is rdev or test
+    API_URL = (
+        "https://api.cellxgene.dev.single-cell.czi.technology"
+        if os.environ.get("DEPLOYMENT_STAGE") in ["test", "rdev"]
+        else os.getenv("API_URL")
+    )
+
     s3_uris = dict()
-    for dataset in get_business_logic().get_all_mapped_datasets():
-        if (
-            (dataset.metadata is not None)
-            and (dataset.metadata.assay is not None)
-            and (dataset.metadata.is_primary_data == "PRIMARY")
-            and (dataset.metadata.organism is not None)
-            and any(assay.ontology_term_id in INCLUDED_ASSAYS for assay in dataset.metadata.assay)
-            and len(dataset.metadata.organism) < 2
-        ):
-            s3_uri = next(
-                a.uri
-                for a in dataset.artifacts
-                if a.type == DatasetArtifactType.H5AD and a.get_file_name() == "local.h5ad"
-            )
-            s3_uris[dataset.dataset_id.id] = s3_uri
+    if API_URL:
+        dataset_metadata_url = f"{API_URL}/dp/v1/datasets/index"
```

Comment:
Can Discover API be called instead? Thinking that GE as more of an external consumer of the data, rather than a part of the Data Portal web app (which should be the only caller of the DP API).

Reply:
Also some significant changes are coming to the Discover API for retrieving datasets, so we might hold off on the PR until that's available.

Reply:
It appears that the curation API datasets endpoint does not provide the `s3_uri` information. This is required for the pipeline, so I think we need to stick with the `datasets/index` endpoint.